test.js

```diff
@@ -71,40 +75,69 @@ test('check if different elements ready inside different targets with same selec
 	t.is(element2.id, 'unicorn2');
 });
 
+test('check if element ready after dom loaded', async t => {
+	const elementCheck = elementReady('#bio', {
+		stopOnDomReady: true
+	});
+
+	// The element will be added eventually, but we're not around to wait for it
+	setTimeout(() => {
+		const element = document.createElement('p');
+		element.id = 'bio';
+		document.body.append(element);
+	}, 50000);
+
+	const element = await elementCheck;
+	t.is(element, undefined);
+});
+
+test('check if element ready before dom loaded', async t => {
+	const element = document.createElement('p');
+	element.id = 'history';
+	document.body.append(element);
+
+	const elementCheck = elementReady('#history', {
+		stopOnDomReady: true
+	});
+
+	t.is(await elementCheck, element);
+});
+
 test('ensure only one promise is returned on multiple calls passing the same selector', t => {
-	const elementCheck = elementReady('#unicorn');
+	const elementCheck = elementReady('#not-found', {stopOnDomReady: false});
 
 	for (let i = 0; i <= 10; i++) {
-		if (elementReady('#unicorn') !== elementCheck) {
+		if (elementReady('#not-found', {stopOnDomReady: false}) !== elementCheck) {
 			t.fail();
 		}
 	}
 
 	t.pass();
 });
 
-test('check if wait can be canceled', async t => {
-	const elementCheck = elementReady('#dofle');
+test('check if wait can be stopped', async t => {
+	const elementCheck = elementReady('#dofle', {stopOnDomReady: false});
 
 	await delay(200);
-	elementCheck.cancel();
+	elementCheck.stop();
 
 	await delay(500);
 	const element = document.createElement('p');
 	element.id = 'dofle';
 	document.body.append(element);
 
-	await t.throwsAsync(elementCheck, PCancelable.CancelError);
+	t.is(await elementCheck, undefined);
 });
 
-test('ensure different promises are returned on second call with the same selector when first was canceled', async t => {
-	const elementCheck1 = elementReady('.unicorn');
+test('ensure different promises are returned on second call with the same selector when first was stopped', async t => {
+	const elementCheck1 = elementReady('.unicorn', {stopOnDomReady: false});
+
+	elementCheck1.stop();
 
-	elementCheck1.cancel();
+	const elementCheck2 = elementReady('.unicorn', {stopOnDomReady: false});
 
-	const elementCheck2 = elementReady('.unicorn');
+	t.is(await elementCheck1, undefined);
```

Comment:
Does this do what you wanted? Or does it no longer apply?

Reply:
If you mean resolving to `undefined` when `.stop()` is called, then yes, it's what I wanted.